src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs

```diff
@@ -2435,33 +2430,6 @@ internal override Symbol RemapSymbolIfNecessaryCore(Symbol symbol)
                     }
                     break;
 
-                case RecordStructDeclarationSyntax recordStructDeclaration when TryGetSynthesizedRecordConstructor(recordStructDeclaration) is SynthesizedRecordConstructor ctor:
-                    switch (declaredSymbol.Kind)
-                    {
-                        case SymbolKind.Method:
```

Comment:
>case SymbolKind.Method: [](http://example.com/codeflow?start=24&length=23)

I think we should preserve the difference for this case.

Reply:
I'm not seeing why (here or below). We get the same behavior. Do you mean just as an optimization?

---
In reply to: [616734766](https://github.com/dotnet/roslyn/pull/52702#discussion_r616734766) [](ancestors = 616734766)

Reply:
Oh, I see. This would be necessary if we add assertion in `PrimaryConstructorBaseType` as you suggested... Let me take a look

---
In reply to: [616930880](https://github.com/dotnet/roslyn/pull/52702#discussion_r616930880) [](ancestors = 616930880,616734766)

Reply:
> I'm not seeing why (here or below). We get the same behavior. Do you mean just as an optimization?

I do not think we should look at bases for record structs even when they are present in an error scenario